client/analytics/settings/index.js

```diff
@@ -87,7 +87,23 @@ class Settings extends Component {
 		}
 	}
 
+	/**
+	 * Ensure changes are reflected to parameters on the window as well as
+	 * the config for construction of this component when re-navigating to
+	 * the settings page.
```

Comment:
Previously, these values updated automatically for objects and arrays due to object equality. I think thats the name, but this is what I mean:

```js
var a = { value: 1 };
var b = a;
b.value = 99;
console.log( a.value ); // 99
```
This won't work for strings, so values need to be updated so that navigating away from the Settings page after saving reflects changes.

Reply:
Yes, this is kinda confusing. :confused:  I wonder if we should clone all `wcSettings.wcAdminSettings` properties we use in `settings/config.js` so we keep them immutable.

So from:
```ES6
initialValue: wcSettings.wcAdminSettings.woocommerce_excluded_report_order_statuses || [],
```
to:
```ES6
initialValue: [ ...wcSettings.wcAdminSettings.woocommerce_excluded_report_order_statuses ] || [],
```
@psealock what would you think about this? :point_up_2: It doesn't fix any bug but I think knowing `wcSettings.wcAdminSettings` is ONLY updated in `persistChanges` will make our lives easier in the future.

Reply:
Good idea, this sets a pattern and lets you know the setting on the window is immutable.